query-translator/tests/translate.rs

```diff
@@ -1134,3 +1134,42 @@ fn test_tx_before_and_after() {
                      WHERE `datoms00`.tx < 12345");
     assert_eq!(args, vec![]);
 }
+
+
+#[test]
+fn test_tx_ids() {
+    let schema = prepopulated_typed_schema(ValueType::Double);
+
+    let query = r#"[:find ?tx :where [(tx-ids $ 1000 2000) [[?tx]]]]"#;
+    let SQLQuery { sql, args } = translate(&schema, query);
+    assert_eq!(sql, "SELECT DISTINCT `transaction_log00`.tx AS `?tx` \
+                     FROM `transaction_log` AS `transaction_log00` \
+                     WHERE 1000 <= `transaction_log00`.tx \
+                       AND `transaction_log00`.tx < 2000");
+    assert_eq!(args, vec![]);
+
+    let query = r#"[:find (count ?tx) :where [(tx-ids $ 1000 2000) [?tx ...]]]"#;
+    let SQLQuery { sql, args } = translate(&schema, query);
+    assert_eq!(sql, "SELECT count(`?tx`) AS `(count ?tx)` FROM (\
+                     SELECT DISTINCT `transaction_log00`.tx AS `?tx` \
+                     FROM `transaction_log` AS `transaction_log00` \
+                     WHERE 1000 <= `transaction_log00`.tx \
+                       AND `transaction_log00`.tx < 2000)");
+    assert_eq!(args, vec![]);
+
+    // In practice the following query would be inefficient because of the filter on all_datoms.tx,
+    // but that is what (tx-data) is for.
```

Comment:
That ticket is relevant, but this tx-* log API is a way to avoid that complexity entirely.  That is, we push the optimization problem to the consumer rather than trying to do (decidely non-trivial!) optimization in Mentat proper.